src/analysis/cohort.py

```
@@ -0,0 +1,457 @@
+# ── Cohort retention builder ─────────────────────────────────────
+
+
+def build_cohort_retention(
+    customers: pd.DataFrame,
+    events: pd.DataFrame,
+    periods: int = MAX_PERIODS,
+    core_events_only: bool = True,
+    retention_mode: str = "rolling",
+    min_events: int = 1,
+) -> pd.DataFrame:
+    """Build monthly cohort × period retention table.
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)